pkg/handlers/archive.go

```diff
@@ -152,9 +153,13 @@ func (h *archiveHandler) openArchive(
 		}
 		defer rdr.Close()
 
-		return h.openArchive(ctx, depth+1, rdr, dataOrErrChan)
+		// Note: We're limited in our ability to add file names to the archiveEntryPath here, as the decompressor doesn't have access to a fileName value. Instead, we're adding a generic string to indicate that the file is decompressed. This could be improved.
+		if depth > 0 {
+			archiveEntryPath = archiveEntryPath + "(decompressed " + reader.format.Name() + " file)"
+		}
+		return h.openArchive(ctx, depth+1, archiveEntryPath, rdr, dataOrErrChan)
 	case archiver.Extractor:
-		err := archive.Extract(logContext.WithValue(ctx, depthKey, depth+1), reader, nil, h.extractorHandler(dataOrErrChan))
+		err := archive.Extract(logContext.WithValue(ctx, depthKey, depth+1), reader, nil, h.extractorHandler(archiveEntryPath, dataOrErrChan))
```

Comment:
@mcastorina do you think we need `logContext.WithValue(ctx, depthKey, depth+1)` if we're keeping track of depth with a `[]string`?